src/maliput/routing/route.cc

```diff
@@ -359,5 +364,130 @@ int Route::FindDirectionTowardsLaneSRangeWithStraightPredecessor(const Route::La
+                                                           const api::LaneEndSet& ongoing_lane_end_set,
+                                                           double tolerance) {
+  std::vector<api::RoadPosition> matching_positions_b{};
+  std::copy_if(positions_b.begin(), positions_b.end(), std::back_inserter(matching_positions_b),
+               [&ongoing_lane_end_set](const api::RoadPosition& position) {
+                 for (int i = 0; i < ongoing_lane_end_set.size(); ++i) {
+                   if (position.lane == ongoing_lane_end_set.get(i).lane) {
+                     return true;
+                   }
+                 }
+                 return false;
+               });
+  const auto it = std::find_if(matching_positions_b.begin(), matching_positions_b.end(),
+                               [position_a, tolerance](const api::RoadPosition& position) {
+                                 return IsRoadPositionClose(position_a, position, tolerance);
+                               });
+  return it != matching_positions_b.end() ? std::optional<api::RoadPosition>{*it} : std::nullopt;
+}
+
+}  // namespace
+
+std::vector<std::string> Route::ValidateEndToEndConnectivity() const {
+  std::vector<std::string> errors;
+
+  // The start Phase must have only one start position.
+  if (phases_.front().start_positions().size() != 1u) {
+    errors.push_back("Start Phase::start_positions() does not have one position.");
+  }
+  // The end Phase must have only one end position.
+  if (phases_.back().end_positions().size() != 1u) {
+    errors.push_back("End Phase::end_positions() does not have one position.");
+  }
+
+  const double tolerance = road_network_->road_geometry()->linear_tolerance();
+
+  for (size_t phase_index = 0u; phase_index < phases_.size() - 1; ++phase_index) {
+    const Phase& phase_a = phases_[phase_index];
+    const Phase& phase_b = phases_[phase_index + 1u];
+    // The inter Phase interfaces must have the same count of points.
+    if (phase_a.end_positions().size() != phase_b.start_positions().size()) {
+      std::stringstream ss;
+      ss << "Phases[" << phase_index << "].end_positions().size() is " << phase_a.end_positions().size();
+      ss << "and Phases[" << (phase_index + 1u) << "].start_positions().size() is " << phase_b.start_positions().size();
+      errors.push_back(ss.str());
+    }
+
+    // For each point in the end positions of phase_a, we need to find the correspondent
+    // in the start positions of phase_b.
+    // The topological connectivity logic follows:
+    // - For each point in the end positions in phase_a, do
+    //   - Try to find a point in the start positions of phase_b with the same api::LaneId.
+    //     - If found, assert the points are coincident. Throw otherwise.
+    //   - When not found, identify the api::BranchPoint and the respective ongoing set of LaneEnds.
+    //   - Find which position of the end positions in phase_b belong to the ongoing set of LaneEnds.
```

Comment:
This should be intended. Let's also mention the context in which occurs:

```
    //   - When not found, this means the boundary between phase_a and phase_b involves two different lanes. In this
    //     case, identify the api::BranchPoint and the respective ongoing set of LaneEnds.
    //     - Find which position of the end positions in phase_b belong to the ongoing set of LaneEnds.
```

Reply:
Fixed in [75feba2](https://github.com/maliput/maliput/pull/628/commits/75feba2d628971074be119f12ded545d35d9fd29).